helpers_core.go

```diff
@@ -29,21 +65,51 @@ func SetCreateVnicOptions(rawCreateVnicDetails interface{}) (vnicOpts *baremetal
 		vnicOpts.PrivateIp = privateIp.(string)
 	}
 
-	// Work around for tf bug https://github.com/hashicorp/terraform/issues/13512.
-	// For bool values that are nested in maps, if the value is set to true/false then
-	// it will appear here as "1"/"0". However, if the value is set to "true"/"false"
-	// then it will appear here as "true"/"false". ParseBool() handles both of these cases.
 	assignPublicIp := vnic["assign_public_ip"]
 	if assignPublicIp != nil {
 		vnicOpts.AssignPublicIp = new(bool)
-		*vnicOpts.AssignPublicIp, err = strconv.ParseBool(assignPublicIp.(string))
+		*vnicOpts.AssignPublicIp = assignPublicIp.(bool)
+	}
+
+	skipSourceDestCheck := vnic["skip_source_dest_check"]
+	if skipSourceDestCheck != nil {
+		vnicOpts.SkipSourceDestCheck = new(bool)
+		*vnicOpts.SkipSourceDestCheck = skipSourceDestCheck.(bool)
+	}
+
+	return
+}
+
+func SetUpdateVnicOptions(vnicDetailsList []interface{}) (vnicOpts *baremetal.UpdateVnicOptions) {
+	vnic := vnicDetailsList[0].(map[string]interface{})
```

Comment:
Do we want a len or nil check here?

Reply:
It *shouldn't* be an issue, since that is guaranteed before calling this: in VnicAttachment it's required, and in Instance we check with GetOk. And if create_vnic_details is present, then it has MinItems set to 1 (and MaxItems to 1), so it must have exactly 1 item. I'll add a comment to the method that this is an expectation of that parameter.